modules/terraform/var-file.go

```diff
@@ -117,33 +102,135 @@ func GetVariableAsListFromVarFileE(t *testing.T, fileName string, key string) ([
 		return nil, UnexpectedOutputType{Key: key, ExpectedType: "[]interface {}", ActualType: reflect.TypeOf(variable).String()}
 	}
 
+	resultArray := []string{}
 	for _, item := range variable.([]interface{}) {
 		resultArray = append(resultArray, fmt.Sprintf("%v", item))
 	}
 
 	return resultArray, nil
 }
 
-// GetAllVariablesFromVarFile Parses all data from a provided input file found in VarFile and stores the result in the value pointed to by out
-func GetAllVariablesFromVarFile(t *testing.T, fileName string, out interface{}) {
-	err := GetAllVariablesFromVarFileE(t, fileName, out)
+// GetAllVariablesFromVarFileE Parses all data from a provided input file found in VarFile and returns a map of the
+// variable values where the keys are the var names and the value are the parsed var values.
+func GetAllVariablesFromVarFile(t *testing.T, fileName string) map[string]interface{} {
```

Comment:
Hm, this seems like a less ergonomic interface. In the old version, you could pass in a custom fully-typed struct, and have the HCL parsed into that. Now, your only option is an untyped map...

Reply:
I'll see what I can do, but as you know, the HCL2 parser is significantly less ergonomical when it comes to transforming to structs so not sure if we can make it do that.

Reply:
In the worst case, we might have to ship this version first, and then improve later. I don't think it's ideal that these functions are using the HCL1 parser since that means they do not support HCL2 syntax and that can be a problem for any var-file implemented for tf12+.